common/components/PrintableWallet/index.jsx

```diff
@@ -0,0 +1,246 @@
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import PropTypes from 'prop-types';
+import translate from 'translations';
+import QRCode from 'qrcode';
+import { toDataUrl as makeIdenticon } from 'ethereum-blockies';
+
+import ethLogo from 'assets/images/logo-ethereum-1.png';
+import sidebarImg from 'assets/images/print-sidebar.png';
+import notesBg from 'assets/images/notes-bg.png';
+
+const walletWidth = 680;
+const walletHeight = 280;
+
+export default class PrintableWallet extends Component {
+  static propTypes = {
+    privateKey: PropTypes.string,
+    address: PropTypes.string
+  };
+
+  state = {
+    qrCodePkey: null,
+    qrCodeAddress: null
+  };
+
+  componentDidMount() {
+    // Start generating QR Codes as soon as component is ready
+    const opts = {
+      color: {
+        dark: '#000',
+        light: '#fff'
+      },
+      margin: 0,
+      errorCorrectionLevel: 'H'
+    };
+
+    QRCode.toDataURL(this.props.privateKey, opts, (err, url) => {
```

Comment:
That was my initial thought too, and I had wanted to use the identicon component that already exists. However, the identicon had some issues that didn't work with printing (Used a background image instead of an `img` tag) and it made me realize just how brittle this printing stuff is. So I decided to go for a one-off so that we weren't saddled with this printing workarounds in a generic component.

If you still want it in a component, I'd be happy to oblige, I just wanted to give you some context.